bson.cpp

```diff
@@ -150,6 +157,14 @@ void VariantToBsonConverter::convertElement(bson_t *bson, const char *key, Varia
 		case KindOfRef:
 			convertElement(bson, key, *v.getRefData());
 			break;
+#if HIPPO_HHVM_VERSION >= 31700
+		case KindOfPersistentVec:
+		case KindOfPersistentDict:
+		case KindOfPersistentKeyset:
+		case KindOfVec:
+		case KindOfDict:
+		case KindOfKeyset:
+#endif
```

Comment:
Is there a way to make the error message actually substitute the type's name? Otherwise, "resource" will likely confuse any folks actually using one of these new types.